### PR TITLE
Regulations 3000: Add models and fixture data

### PR DESCRIPTION
**Pull request by willbarton**
This PR is a combination of work by @higs4281, @contolini, and myself.

## Additions

- Django data models for Regulations 3000
- New Wagtail pages and templates for the Regulations 3000 landing page and individual Regulations
- Fixture data for initial population for the data models
- Regdown, a Python Markdown extension that supports labeled paragraphs and block references

## Testing

1. Unittest:
   ```
   tox -e lint-py27 \
         -e lint-py36 \ 
         -e unittest-py27-dj18-wag110-fast \
         -e unittest-py36-dj18-wag110-fast \
         regulations3k
   ```
2. Load test data:
   ```
   django-admin.py loaddata regdata
   ```
3. Create a `RegulationLandingPage` at `/eregulations3k`
4. Create a `RegulationPage` within that landing page and select `12 CFR Part 1005 (Regulation E)` in the `Regulation` field, and give it the slug `1005`.
5. Visit http://localhost:8000/eregulations3k/1005/

## Notes

I can't personally run any of the unittests at the moment because I'm having serious issues with the MySQL-python dependency locally. Both the py27 and py36 tests *should* successfully pass, and did before rebasing. But YMMV.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by willbarton** [comment]
I'll rebase after #4059 is merged to fix the moto failure in Travis.

File: .gitignore [hunk 1]
```diff
@@ -35,7 +35,8 @@ cfgov-refresh/
 ._*
 .Spotlight-V100
 .Trashes
-Icon
+Icon
+
```
[the remark this answers is not in the document]
Comment (willbarton) [reply .gitignore]: No, I don't think it is.

[the remark this answers is not in the document]
Comment (contolini) [reply .gitignore]: Ditto, not intentional. Safe to revert.

File: .gitignore [hunk 2]
```diff
@@ -35,8 +35,7 @@ cfgov-refresh/
 ._*
 .Spotlight-V100
 .Trashes
-Icon
-
+Icon
```
Comment (jimmynotjim) [inline .gitignore]: Please revert this change. 

@chosak Any idea how to keep this in when most of us have a whitespace eliminator set up in our editor of choice?

Comment (jimmynotjim) [reply .gitignore]: Sorry, I realize what's happening now, I had it in reverse in my mind.

@chosak I had the same thing happen when I edited the gitignore locally. The change from https://stackoverflow.com/questions/17556250/how-to-ignore-icon-in-git/21111738#21111738 seems to be converted to new lines when you save the gitignore in an editor. Any idea how we can prevent that?

File: .gitignore [hunk 3]
```diff
@@ -105,7 +104,6 @@ htmlcov/
 nosetests.xml
 coverage.xml
 .snyk
-.python-version
```
Comment (jimmynotjim) [inline .gitignore]: Are you sure you wanted to revert this as well?

Comment (willbarton) [reply .gitignore]: I'm not sure where `.python-version` comes from.

File: .gitignore (hunk 2 above)
Comment (chosak) [reply .gitignore]: Unfortunately no; I see though that the way the change was made is pretty brittle and prone to being removed. Do we need this line? When are these files created -- only when browsing in an image folder on a Mac?

Comment (jimmynotjim) [reply .gitignore]: It seems that way. I have never seen one included before but I also think it's such a standard that every repo I've worked in since starting with git has included the old pattern you had updated. I guess it's worth removing and keeping an eye out for unintended icon files?

**Comment by contolini** [comment]
Alrighty the PWA branch has been brought over. This PR is all of Regulations 3K. And it's a doozy.

**Comment by contolini** [comment]
When I click through to codecov I'm not seeing any files with reduced coverage. Soooo I think this PR is ready for final review and merging.
